### ISSUE TITLE
Segfault in Distributor

### ISSUE THREAD

**Issue by aprokop** [arborx/ArborX#295]
#222 (checked through bisection) seems to have introduced a new bug, that results in segfault when running distributed benchmark with CUDA backend. I am able to reproduce it even running as
```shell
mpirun  -np 2 ./ArborX_DistributedTree.exe --node cuda --values=1 --queries=1 
```
As long as at least one search is performed (does not matter radius or nearest), it segfaults. The backtrace that I've seen points to `doPostsAndWaits` in Distributor, specifically to `MPI_Waitall` call.

@masterleinad I hope you can help me with this.

**Comment by aprokop** [comment]
With @masterleinad's help, the issue was identified in the mode when permutation is not necessary. In this case, `send_buffer_ptr` was set to `exports.data()` which in #222 was changed to live on host.

This problem was never detected as all the testing we do uses CUDA-aware MPI (which, to our slight surprise, even allows mixed CPU-GPU mode).

@dalg24 Do you see any issues with making some of the containers use non CUDA-aware MPI?

**Comment by dalg24** [comment]
> Do you see any issues with making some of the containers use non CUDA-aware MPI?

No that's fine.  My understanding is we have already one (CUDA-10.1-NVCC)

**Comment by aprokop** [comment]
Correction to my previous comment:

Right now, `--with-cuda` option for building OpenMPI in docker is tied to `$BASE`. If `$BASE` is CUDA, we also build CUDA-aware MPI. Thus, while we actually have builds that don't have CUDA-aware MPI, those builds also don't have CUDA on, and thus we never run into the issue.

**Comment by aprokop** [comment]
Fixed in #296.
